interfaces/cython/cantera/thermo.pyx

```diff
@@ -809,15 +809,19 @@ cdef class ThermoPhase(_SolutionBase):
         self.state = original_state
         return arr
 
-    def set_equivalence_ratio(self, phi, fuel, oxidizer, basis='mole'):
+    def set_equivalence_ratio(self, phi, fuel, oxidizer, basis='mole', fraction=None, diluent=None):
```

Comment:
Changing the order of the arguments would possibly break existing code, so I don't think we should do that. I'd be in favor of making everything after `basis` into a keyword-only argument, I think if people are listing that many arguments they should be using keywords anyways.

Reply:
Good point. I'd still advocate for reversing the order, i.e. `diluent` before `fraction`.

Reply:
My reasoning for the current argument order was to mirror the existing structure (first how to mix and then what to mix). At the moment, it is first phi, i.e. a type of mixing ratio, and then the reactant compositions. This is now followed by the dilution fraction and then by the diluent composition. But I don't feel strongly either way.

Reply:
I still believe it makes sense to have `fraction` last as it can reference fuel, oxidizer and diluent. Regarding @bryanwweber’s comment, I believe it makes sense to force keyword arguments. See explanation here: [vegardstikbakke.com/python-keyword-only/](https://www.vegardstikbakke.com/python-keyword-only/)